pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -293,6 +294,23 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	// the attached volumes field. This function ensures that we sync with
 	// the actual status.
 	adc.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+
+	if _, exists := node.Annotations[volumehelper.ControllerManagedAttachAnnotation]; exists {
+		// Newly added node is managed by controller, ensure all pods are up to date.
+		adc.processPodsForNode(nodeName)
+	}
+}
+
+func (adc *attachDetachController) processPodsForNode(nodeName types.NodeName) {
+	pods, err := adc.podLister.List(labels.Everything())
```

Comment:
This is a problem at startup especially. For every node in the cluster it will do a list of the pods. I notice we do lots of list alls though (like in daemonset controller), so I'm not sure if it's a big deal or not.